string direction
float32 distance